### DISCUSSION TITLE
Can Quarto output the commands it runs as it renders?

### DISCUSSION THREAD

**Discussion by cskeeters** [quarto-dev/quarto-cli#10873] [Q&A] [answered]
### Description

I'm just wondering if there is a way to run `quarto render` so that the commands issued (`pandoc` `xelatex`) and all of their parameters will be added to the log.  `--log info` was insufficient.

I just want to understand better how the document is being processed and be able to reproduce and debug the pipeline if necessary.

**Comment by cscheid** [comment, marked as answer]
> all of their parameters
>
> I just want to understand better how the document is being processed and be able to reproduce

I understand the motivation, and I hear you. This is unlikely to be easy outside of Quarto, because of the number of filters and post-processing steps we perform.

> and debug the pipeline if necessary.

I think we should consider something like that in a particularly verbose mode, but I will warn you ahead of time that this is going to be _a lot of information_, and much of what we pass Pandoc happens through content in temporary files that are erased after the execution happens.

In the limit, I think what you're asking for needs to be addressed after we fix https://github.com/quarto-dev/quarto-cli/issues/6518 (which is going to happen for sure, but not anytime soon.)
